Mathlib/CategoryTheory/DiscreteCategory/SumsProducts.lean

```diff
@@ -0,0 +1,104 @@
+/-
+Copyright (c) 2025 Robin Carlier. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Robin Carlier
+-/
+import Mathlib.CategoryTheory.DiscreteCategory.Basic
+import Mathlib.CategoryTheory.Sums.Basic
+import Mathlib.CategoryTheory.Products.Basic
+/-! # Sums and products of discrete categories.
+
+This file shows that binary products and binary sums of discrete categories
+are also discrete, both in the form of explicit equivalences and through the
+`IsDiscrete` typeclass.
+
+## Main declarations
+
+* `Discrete.productEquiv`: The equivalence of categories between `Discrete (J × K)`
+  and `Discrete J × Discrete K`
+* `Discrete.sumEquiv`: The equivalence of categories between `Discrete (J ⊕ K)`
+  and `Discrete J ⊕ Discrete K`.
+* `IsDiscrete.prod`: an `IsDiscrete` instance on the product of two discrete categories.
+* `IsDiscrete.sum`: an `IsDiscrete` instance on the sum of two discrete categories.
+
+-/
+universe v₁ u₁
+
+namespace CategoryTheory
+
+namespace Discrete
+
+/-- The discrete category on a product is equivalent to the product of the
+discrete categories. -/
+@[simps!]
+def productEquiv {J K : Type*} : Discrete (J × K) ≌ Discrete J × Discrete K where
+  functor := Discrete.functor <| fun ⟨j, k⟩ ↦ ⟨.mk j, .mk k⟩
+  inverse := {
+    obj := fun ⟨x, y⟩ ↦ .mk (⟨x.as, y.as⟩)
+    map := fun ⟨f₁, f₂⟩ ↦ eqToHom (by discrete_cases; dsimp; rw [f₁, f₂]) }
+  unitIso := NatIso.ofComponents (fun _ ↦ Iso.refl _)
+  counitIso := NatIso.ofComponents (fun _ ↦ Iso.refl _)
+
+/-- The discrete category on a sum is equivalent to the sum of the
+discrete categories. -/
+@[simps!]
+def sumEquiv {J K : Type u₁} : Discrete (J ⊕ K) ≌ Discrete J ⊕ Discrete K where
```

Comment:
Is it possible to allow `J` and `K` to be in different universes?

Reply:
I looked into this also, and it seems that the sum is not defined in this case, as it is defined to live in the same universe. I don't know if one should generalize that file or not.

Reply:
Looks like trying to generalize `CategoryTheory.Sums.Basic` to allow for different universes is not that straightforward, so I’m leaving this as is.